### PR TITLE
pythonlib: Fix ResourceWarning in libgrass_interface_generator/ctypesgen/version.py

### PR DESCRIPTION
**Pull request by arohanajit**
This pull request addresses the recurring `ResourceWarning` about an unclosed file in the version.py script. The following changes have been made:

- Modified the version() function to use a context manager for handling the file opened for /dev/null, ensuring proper closure.
- Updated read_file_version() and write_version_file() functions to also use context managers for file operations.
- Updated the write_version_file() method to use `Path(VERSION_FILE).write_text(v)` which fixes the FURB103 warning generated further as pointed out here: https://github.com/OSGeo/grass/pull/4285#discussion_r1749008784

These changes should resolve the ResourceWarnings in this particular case

### CODE REVIEW

File: python/libgrass_interface_generator/ctypesgen/version.py [hunk 1]
```diff
@@ -60,9 +60,9 @@ def compatible(v0, v1):
 def write_version_file(v=None):
     if v is None:
         v = version()
-    f = open(VERSION_FILE, "w")
-    f.write(v)
-    f.close()
+    # with open(VERSION_FILE, "w") as f:
+    #     f.write(v)
+    Path(VERSION_FILE).write_text(v)
```
Comment (echoix) [inline python/libgrass_interface_generator/ctypesgen/version.py:65]:
```suggestion
    Path(VERSION_FILE).write_text(v)
```

**Review by echoix — DISMISSED** [review]

**Review by nilason — COMMENTED** [review]
We shouldn't waste time on non-crucial fixes here on ctypesgen, with next update from upstream, they will be null-and-void (see https://github.com/OSGeo/grass/blob/main/python/libgrass_interface_generator/README.md).

**Comment by echoix** [comment]
Do you know if upstream is planning to make a release real soon?

**Comment by nilason** [comment]
> Do you know if upstream is planning to make a release real soon?

It looks like it's pretty much up to me unfortunately, as I'm the only active with enough access rights. So, probably not "real" soon. Still, these kind of changes are not urgent.

**Comment by echoix** [comment]
Would you want these changes in a PR in the upstream repo so they could be there when ready?

**Comment by arohanajit** [comment]
@echoix just wanted to confirm if these are still required. If not, I'll close this PR

**Comment by echoix** [comment]
We might as well close them. But they still exist to search again at some point. It's kinda sad, but otherwise it'll end up like one of the 130 other pending PRs

**Pull request closed without merging**